### ISSUE TITLE
ethereumjs-util: "This method only supports Buffer"

### ISSUE THREAD

**Issue by frangio** [nomiclabs/hardhat#1885]
Seeing this error when trying to update our deps:

```
Error: This method only supports Buffer but input was: 0xac...
    at Object.exports.assertIsBuffer (/tmp/2021-09-15/19h-42m-SvG/node_modules/ethereumjs-util/src/helpers.ts:21:11)
    at Object.exports.privateToPublic (/tmp/2021-09-15/19h-42m-SvG/node_modules/ethereumjs-util/src/account.ts:275:3)
    at exports.privateToAddress (/tmp/2021-09-15/19h-42m-SvG/node_modules/ethereumjs-util/src/account.ts:285:26)
    at /tmp/2021-09-15/19h-42m-SvG/node_modules/@nomiclabs/hardhat-truffle5/src/index.ts:140:41
    at Array.map (<anonymous>)
    at describeContract (/tmp/2021-09-15/19h-42m-SvG/node_modules/@nomiclabs/hardhat-truffle5/src/index.ts:139:11)
```

https://github.com/nomiclabs/hardhat/blob/21ab17bc586e061315be45954aeb495f557972dd/packages/hardhat-truffle5/src/index.ts#L140

`acc.privateKey` is a string and `privateToAddress` expects a Buffer.

**Comment by cwsnt** [comment]
Hi @frangio , any workaround for this? Facing this issue

**Comment by swaylock** [comment]
Same.

**Comment by swaylock** [comment]
@frangio in the discord, you mention in discord: "You can try downgrading (Truffle, not our plugin)". Would you mind elaborating? Thanks!

**Comment by frangio** [comment]
As far as I know this is a bug in the hardhat plugin and there is no workaround.

**Comment by swaylock** [comment]
The author of https://github.com/OpenZeppelin/openzeppelin-contracts/issues/2880 (same issue, but in OpenZeppelin) suggests the bug may be hardware/OS based?

Would be nice to eliminate that. Can anyone confirm whether the following works for them:

`git clone git@github.com:OpenZeppelin/openzeppelin-contracts.git
cd openzeppelin-contracts
npm install
npm run test`

And if so, indicate computer & OS?

**Comment by itinance** [comment]
I am the author of the referred https://github.com/OpenZeppelin/openzeppelin-contracts/issues/2880 and can report that this issue seems to be solved right now.

Since I kept the older folder of "git clone"-output of openzeppelin I could check the difference between the state when this issue was happening and the state of today:

`/usr/bin/diff /tmp/openzeppelin-contracts/package.json /tmp/o/openzeppelin-contracts/package.json`

```
53c53
<     "@nomiclabs/hardhat-truffle5": "^2.0.1",
---
>     "@nomiclabs/hardhat-truffle5": "^2.0.0",
70c70
<     "hardhat": "^2.6.4",
---
>     "hardhat": "^2.0.6",
72c72
72c72
<     "keccak256": "^1.0.3",
---
>     "keccak256": "^1.0.2",
75,78c75,78
<     "merkletreejs": "^0.2.24",
<     "micromatch": "^4.0.4",
<     "prettier": "^2.4.1",
<     "prettier-plugin-solidity": "^1.0.0-beta.18",
---
>     "merkletreejs": "^0.2.13",
>     "micromatch": "^4.0.2",
>     "prettier": "^2.3.0",
>     "prettier-plugin-solidity": "^1.0.0-beta.16",
81,85c81,85
<     "solidity-ast": "^0.4.26",
<     "solidity-coverage": "^0.7.17",
<     "solidity-docgen": "^0.5.14",
<     "web3": "^1.5.3",
<     "yargs": "^17.2.0"
---
>     "solidity-ast": "^0.4.25",
>     "solidity-coverage": "^0.7.11",
>     "solidity-docgen": "^0.5.3",
>     "web3": "^1.3.0",
>     "yargs": "^16.2.0"
```

(I left out all lines that belongs to unit testing and linting here)

I was running successfully 

```
git clone git@github.com:OpenZeppelin/openzeppelin-contracts.git
cd openzeppelin-contracts
npm install
npm run test
```

both on Intel based Macs (where this issue never appeared for me) and ARM based Mac M1, where this issue was happening reproducable.

(/cc @swaylock)

**Comment by tina1998612** [comment]
@itinance still not working for me after updating these dependencies

**Comment by tina1998612** [comment]
@itinance I tried Docker and still the same issue. 😢

## Dockerfile: 
```
FROM node:14-alpine
COPY . /usr/src/app
WORKDIR /usr/src/app
RUN apk add git;
RUN yarn install --non-interactive --frozen-lockfile
COPY $PWD/docker/entrypoint.sh /usr/local/bin
ENTRYPOINT ["/bin/sh", "/usr/local/bin/entrypoint.sh"]
```

## entrypoint.sh
```
#!/bin/sh

# Change to the correct directory
cd /usr/src/app;

# Run hardhat coverage
npm run coverage
```

## Error
```
Error in plugin solidity-coverage: Error: This method only supports Buffer but input was: 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
```

**Comment by frangio** [comment]
This is not related to platform. It's quite clear in the stack trace and ethereumjs-util changelog: the privateToAddress/privateToPublic functions now expect a Buffer but Hardhat passes a string.

https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/util/CHANGELOG.md#enforce-buffer-inputs-for-account-methods

@fvictorio Note that this bug is making the latest version of the truffle5 plugin unusable.

**Comment by tina1998612** [comment]
@frangio so the temporary solution is to downgrade truffle5 plugin?

**Comment by FortisFortuna** [comment]
Same issue

**Comment by fvictorio** [comment]
Fixed in `@nomiclabs/hardhat-truffle4@2.0.2` and `@nomiclabs/hardhat-truffle5@2.0.2`!
